nilearn/input_data/nifti_labels_masker.py

```diff
@@ -184,7 +204,107 @@ def __init__(self, labels_img, background_label=0, mask_img=None,
+            # Number of regions excluding the background
+            number_of_regions = np.sum(np.unique(labels_image_data) !=
+                                       self.background_label)
+            # Basic safety check to ensure we have as many labels as we
+            # have regions (plus background).
+            if(self.labels is not None and
+               len(self.labels) != number_of_regions + 1):
+                raise ValueError(("Mismatch between the number of provided "
+                                  "labels ({0}) and the number of regions "
+                                  "in provided label image ({1})").format(
+                                      len(self.labels), number_of_regions + 1))
+            self._report_content['number_of_regions'] = number_of_regions
+
+            label_values = np.unique(labels_image_data)
+            label_values = label_values[label_values != self.background_label]
+            columns = ['label value', 'region name', 'size (in mm^3)',
+                       'relative size (in %)']
+            if self.labels is None:
+                columns.remove('region name')
+            regions_summary = {c:[] for c in columns}
+            for label in label_values:
+                regions_summary['label value'].append(label)
+                if self.labels is not None:
+                    regions_summary['region name'].append(self.labels[label])
+                size = len(labels_image_data[labels_image_data == label])
+                voxel_volume = np.abs(np.linalg.det(
+                                    labels_image_affine[:3, :3]))
+                regions_summary['size (in mm^3)'].append(round(
+                                        size * voxel_volume))
+                regions_summary['relative size (in %)'].append(round(
+                    size / len(labels_image_data[labels_image_data != 0]) * 100,
+                    2))
+            self._report_content['summary'] = regions_summary
+
+            img = self._reporting_data['img']
+            # If we have a func image to show in the report, use it
+            if img is not None:
+                dim = image.load_img(img).shape
+                if len(dim) == 4:
+                    # compute middle image from 4D series for plotting
+                    img = image.index_img(img, dim[-1] // 2)
+                display = plotting.plot_img(img,
+                                            black_bg=False,
+                                            cmap='CMRmap_r')
+                display.add_contours(labels_image,
+                                     filled=False,
+                                     linewidths=3)
+
+            # Otherwise, simply plot the ROI of the label image
+            # and give a warning to the user
+            else:
+                msg = ("No image provided to fit in NiftiLabelsMasker. "
+                       "Plotting ROI of label image for reporting.")
```

Comment:
I think we could be a bit clearer in this warning. Maybe :point_down: 

```suggestion
                msg = ("No image provided to fit in NiftiLabelsMasker. "
                       "Plotting ROIs of label image on the MNI152Template for reporting.")
```

Reply:
Thanks!